A physician is evaluating a new diet for her patients with a family history of heart disease. To test the effectiveness of this diet, 16 patients are placed on the diet for 6 months. Their weights and triglyceride levels are measured before and after the study, and the physician wants to know if either set of measurements has changed.

In [4]:
import pandas as pd
import scipy.stats as stats

In [5]:
dietstudy=pd.read_csv("C:\Users\hp\Downloads\python\dietstudy.csv")
dietstudy.head()

patid  age  gender  tg0  tg1  tg2  tg3  tg4  wgt0  wgt1  wgt2  wgt3  wgt4
0      1   45    Male  180  148  106  113  100   198   196   193   188   192
1      2   56    Male  139   94  119   75   92   237   233   232   228   225
2      3   50    Male  152  185   86  149  118   233   231   229   228   226
3      4   46  Female  112  145  136  149   82   179   181   177   174   172
4      5   64    Male  156  104  157   79   97   219   217   215   213   214

In [6]:
a=dietstudy.tg0.mean()
b=dietstudy.tg4.mean()
a-b

14.0625

In [7]:
dietstudy.wgt0.mean()-dietstudy.wgt4.mean()

8.0625

In [9]:
stats.ttest_rel(a=dietstudy.tg0,b=dietstudy.tg4) 
#stats.ttest_rel(a=dietstudy.wgt0,b=dietstudy.wgt4) 

Ttest_relResult(statistic=1.2000008533342437, pvalue=0.24874946576903698)

###### Conclusions which can be drawn: Since the significance value for change in weight is less than 0.01, you can conclude that the average loss of 8.06 pounds per patient is not due to chance variation, and can be attributed to the diet.

However, the significance value greater than 0.24 for change in triglyceride level shows the diet did not significantly reduce their triglyceride levels.

An analyst at a department store wants to evaluate a recent credit card promotion. To this end, 500 cardholders were randomly selected. Half received an ad promoting a reduced interest rate on purchases made over the next three months, and half received a standard seasonal ad. Is the promotion effective to increase sales?

In [10]:
creditpromo=pd.read_csv("C:\Users\hp\Downloads\python\creditpromo.csv")
creditpromo.head()

id         insert      dollars
0   148       Standard  2232.771979
1   572  New Promotion  1403.807542
2   973       Standard  2327.092181
3  1096       Standard  1280.030541
4  1541  New Promotion  1513.563200

In [11]:
creditpromo.groupby('insert').mean()

id      dollars
insert                               
New Promotion  64998.244  1637.499983
Standard       69003.000  1566.389031

In [12]:
crst=creditpromo[creditpromo['insert'] =='Standard']
crst.count()

id         250
insert     250
dollars    250
dtype: int64

In [13]:
crnp=creditpromo[creditpromo['insert'] =='New Promotion']
crnp.count()

id         250
insert     250
dollars    250
dtype: int64

In [14]:
stats.ttest_ind(a=crst.dollars,b=crnp.dollars)

Ttest_indResult(statistic=-2.2604227264649963, pvalue=0.024225996894147814)

###### Conclusion: With 95% signifiance we can say that Promotion has increased the sales.

An experiment is conducted to study the hybrid seed production of bottle gourd under open field conditions. The main aim of the investigation is to compare natural pollination and hand pollination. The data are collected on 10 randomly selected plants from each of natural pollination and hand pollination. The data are collected on fruit weight (kg), seed yield/plant (g) and seedling length (cm). (Data set: pollination.csv)
a. Is the overall population of Seed yield/plant (g) equals to 200?
b. Test whether the natural pollination and hand pollination under open field conditions are equally effective or are significantly different.


In [16]:
pol=pd.read_csv("C:\Users\hp\Downloads\python\pollination.csv")
pol.head()
pol['syfw']=pol['Seed_Yield_Plant']
stats.ttest_1samp(a=pol.syfw,popmean=200)

Ttest_1sampResult(statistic=-2.3009121248548645, pvalue=0.032891040921283025)

In [17]:
pol.head()

Group  Fruit_Wt  Seed_Yield_Plant  Seedling_length    syfw
0  Natural      1.85            147.70            16.86  147.70
1  Natural      1.86            136.86            16.77  136.86
2  Natural      1.83            149.97            16.35  149.97
3  Natural      1.89            172.33            18.26  172.33
4  Natural      1.80            144.46            17.90  144.46

In [20]:
pol['Seed_Yield_Plant'].mean()

180.8035

###### From pop we can see that mean<200

Test whether the natural pollination and hand pollination under open field conditions are equally effective or are significantly diferent

In [21]:
pol.head()

Group  Fruit_Wt  Seed_Yield_Plant  Seedling_length    syfw
0  Natural      1.85            147.70            16.86  147.70
1  Natural      1.86            136.86            16.77  136.86
2  Natural      1.83            149.97            16.35  149.97
3  Natural      1.89            172.33            18.26  172.33
4  Natural      1.80            144.46            17.90  144.46

In [25]:
polna=pol.where(pol.Group=='Natural').dropna()
polhand=pol.where(pol.Group=='Hand').dropna()

In [27]:
polna.Seed_Yield_Plant.mean()

146.009

In [28]:
polhand.Seed_Yield_Plant.mean()

215.598

In [26]:
stats.ttest_ind(a=polna.Seed_Yield_Plant,b=polhand.Seed_Yield_Plant)

Ttest_indResult(statistic=-13.958260515902547, pvalue=4.2714815854843853e-11)

###### From the results we can see that hand pollination has better results in comparison to natural pollination

In [32]:
un25=dvdplayer[dvdplayer['agegroup'] =='Under 25']

An electronics firm is developing a new DVD player in response to customer requests. Using a prototype, the marketing team has collected focus data for different age groups viz. Under 25; 25-34; 35-44; 45-54; 55-64; 65 and above. Do you think that consumers of various ages rated the design differently? (Data set: dvdplayer.csv).

In [30]:
dvdplayer=pd.read_csv("C:\Users\hp\Downloads\python\dvdplayer.csv")
dvdplayer.head()

agegroup   dvdscore
0  65 and over  38.454803
1        55-64  17.669677
2  65 and over  31.704307
3  65 and over  25.924460
4     Under 25  30.450007

In [33]:

un25.mean()

dvdscore    28.749228
dtype: float64

In [34]:
un2534=dvdplayer[dvdplayer['agegroup'] =='25-34']
un2534.mean()

dvdscore    31.67801
dtype: float64

In [35]:
un3544=dvdplayer[dvdplayer['agegroup'] =='35-44']
un3544.mean()

dvdscore    37.018058
dtype: float64

In [36]:
un4554=dvdplayer[dvdplayer['agegroup'] =='45-54']
un4554.mean()

dvdscore    39.118369
dtype: float64

In [37]:
un5564=dvdplayer[dvdplayer['agegroup'] =='55-64']
un5564.mean()

dvdscore    28.447336
dtype: float64

In [38]:
un65=dvdplayer[dvdplayer['agegroup'] =='65 and over']
un65.mean()

dvdscore    28.002791
dtype: float64

In [39]:
stats.f_oneway(un25.dvdscore,un2534.dvdscore,un3544.dvdscore,un4554.dvdscore,un5564.dvdscore,un65.dvdscore)

F_onewayResult(statistic=6.992526962676518, pvalue=3.087324905679639e-05)

###### So with 99.99% we can say that consumers of different age groups rated the design differently.

A survey was conducted among 2800 customers on several demographic characteristics. Working status, sex, age, age-group, race, happiness, no. of child, marital status, educational qualifications, income group etc. had been captured for that purpose. (Data set: sample_survey.csv).
a. Is there any relationship in between labour force status with marital status?
b. Do you think educational qualification is somehow controlling the marital status?
c. Is happiness is driven by earnings or marital status?